build/SharedFx.targets

```diff
@@ -121,6 +127,137 @@
+    <Error Text="Expected to resolve a single clr jit assembly but instead resolved @(ClrJitAssembly->Count()) with identities %(ClrJitAssembly.Identity)" Condition="'@(ClrJitAssembly->Count())' != 1" />
+
+    <!-- Gather details on published assemblies -->
+    <MSBuild Projects="$(SharedFxWorkDirectory)SharedFx.csproj"
+      Targets="GetPublishAssemblies"
+      Properties="RuntimeIdentifier=$(SharedFxRID);SelfContained=false" >
+      <Output TaskParameter="TargetOutputs" ItemName="PublishAssemblies" />
+    </MSBuild>
+
+    <ItemGroup>
+      <IgnoredAssemblies Include="@(PublishAssemblies)" Condition="'%(AssetType)' == 'native' OR '%(AssetType)' == 'resources'" />
+      <_AssembliesToCrossgen Include="$(SharedFxOutputDirectory)**\*.dll" />
+    </ItemGroup>
+
+    <!-- Compute the intersection of crossgen candidates and native/resources assemblies as the set of assemblies to skip crossgen -->
+    <CreateItem Include="@(_AssembliesToCrossgen)" Condition="'%(Filename)' != ''and '@(IgnoredAssemblies)' != ''">
+      <Output TaskParameter="Include" ItemName="AssembliesToRemove"/>
+    </CreateItem>
+
+    <!-- Resolve list of assemblies to crossgen -->
+    <ItemGroup>
+      <AssembliesToCrossgen Include="@(_AssembliesToCrossgen)">
+        <Source>%(FullPath)</Source>
+        <Rsp>$(SharedFxCrossGenRspDirectory)%(RecursiveDir)%(Filename).rsp</Rsp>
+        <SymbolsRsp>$(SharedFxCrossGenRspDirectory)%(RecursiveDir)%(Filename).symbols.rsp</SymbolsRsp>
+        <Destination>$(SharedFxCrossGenDirectory)%(RecursiveDir)%(Filename)%(Extension)</Destination>
+        <Symbols>$(SharedFxCrossGenDirectory)%(RecursiveDir)</Symbols>
+      </AssembliesToCrossgen>
+      <AssembliesToCrossgen Remove="@(AssembliesToRemove)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="CrossGenAssemblies"
+    DependsOnTargets="PrepareForCrossGen"
+    Inputs="@(AssembliesToCrossgen)"
+    Outputs="%(AssembliesToCrossgen.Destination)">
+    <ItemGroup>
+      <CrossGenArgs Include="-nologo" />
+      <CrossGenArgs Include="-readytorun" />
+      <CrossGenArgs Include="-in %(AssembliesToCrossgen.Source)" />
+      <CrossGenArgs Include="-out %(AssembliesToCrossgen.Destination)" />
+      <CrossGenArgs Include="-app_paths $(SharedFxPublishDirectory)" />
+      <CrossGenArgs Include="-platform_assemblies_paths $(SharedFxCrossGenToolDirectory)" />
+      <CrossGenArgs Include="-JITPath %(ClrJitAssembly.FullPath)" />
+    </ItemGroup>
+
+    <MakeDir Directories="$([System.IO.Path]::GetDirectoryName('%(AssembliesToCrossgen.Rsp)'))" />
+    <MakeDir Directories="$([System.IO.Path]::GetDirectoryName('%(AssembliesToCrossgen.Destination)'))" />
+    <WriteLinesToFile File="%(AssembliesToCrossgen.Rsp)" Lines="@(CrossGenArgs)" Overwrite="true" />
+
+    <Exec Command="$(SharedFxCrossGenToolDirectory)$(CrossGenTool) @%(AssembliesToCrossgen.Rsp)" EnvironmentVariables="COMPlus_PartialNGen=0" />
```

Comment:
Consider also adding `StandardOutputImportance="Normal"`. Makes `-verbose:minimal` actually minimal. You can still see all the crossgen output for `-verbose:normal`